x-pack/test/performance/services/performance.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable no-console */
+
+import Url from 'url';
+import { inspect } from 'util';
+import apm, { Outcome, Span, Transaction } from 'elastic-apm-node';
+import { setTimeout } from 'timers/promises';
+import playwright, { ChromiumBrowser, Page, BrowserContext } from 'playwright';
+import { FtrService, FtrProviderContext } from '../ftr_provider_context';
+type StorageState = Awaited<ReturnType<BrowserContext['storageState']>>;
+
+apm.start({
+  secretToken: 'Q5q5rWQEw6tKeirBpw',
+  serverUrl: 'https://2fad4006bf784bb8a54e52f4a5862609.apm.us-west1.gcp.cloud.es.io:443',
+  serviceName: 'functional test runner',
+});
+
+interface StepCtx {
+  page: Page;
+}
+type StepFn = (ctx: StepCtx) => Promise<void>;
+type Steps = Array<{ name: string; fn: StepFn }>;
+
+export class PerformanceTestingService extends FtrService {
+  private readonly config = this.ctx.getService('config');
+  // private readonly es = this.ctx.getService('es');
+  private readonly lifecycle = this.ctx.getService('lifecycle');
+  private browser: ChromiumBrowser | undefined;
+  private storageState: StorageState | undefined;
+  private currentSpanStack: Array<Span | null> = [];
+  private currentTransaction: Transaction | undefined | null;
+
+  constructor(ctx: FtrProviderContext) {
+    super(ctx);
+
+    this.lifecycle.beforeTests.add(async () => {
+      await this.withTransaction('Journey setup', async () => {
+        await this.getStorageState();
+      });
+    });
+
+    this.lifecycle.cleanup.add(async () => {
+      apm.flush();
+      await setTimeout(5000);
+      await this.browser?.close();
+    });
+  }
+
+  private endTransaction(result: string) {
```

Comment:
Looks like you inlined `startTransaction`, but my stance stands for `endTransaction`, `startSpan`, and `endSpan` too.